File: TestVs2015/TestCpp/TestCpp.py
from vs2015 import *

def CompareList(lhs, rhs):
	if len(lhs) != len(rhs):
		return False
	for i in range(len(lhs)):
		if lhs[i] != rhs[i]:
			return False
	return True
	
def CompareDictionary(lhs, rhs):
	if len(lhs) != len(rhs):
		return False
	for k in lhs:
		if k not in rhs:
			return False
		if lhs[k] != rhs[k]:
			return False
	return True
	
convertToInt = lambda x : int(x.Value)
	
v = LoadVector(AsVariable('v'), convertToInt)
print str(CompareList(v, [10, 11, 12, 13, 14])) + " : LoadVector"

l = LoadList(AsVariable('l'), convertToInt)
print str(CompareList(l, [10, 11, 12, 13, 14])) + " : LoadList"

u = LoadUMap(AsVariable('u'), convertToInt, convertToInt)
print str(CompareDictionary(u, {1: 11, 2: 22, 3: 33})) + " : LoadUMap"

m = LoadMap(AsVariable('m'), convertToInt, convertToInt)
print str(CompareDictionary(m, {1: 11, 2: 22, 3: 33})) + " : LoadMap"

s1 = LoadString(AsVariable('s1'))
print str(s1 == 'hello') + " : LoadString"

s2 = LoadWString(AsVariable('s2'))
print str(s2 == 'world') + " : LoadWString"

s3 = LoadCharArray(AsVariable('s3'), 255)
print str(s3 == 'a small') + " : LoadCharArray"

s4 = LoadWCharArray(AsVariable('s4'), 255)
print str(s4 == 'good thing') + " : LoadWCharArray"

s5 = LoadCharArray(AsVariable('s5'), 255)
print str(s5 == 'python') + " : LoadCharArray"

s6 = LoadWCharArray(AsVariable('s6'), 255)
print str(s6 == 'c sharp') + " : LoadWCharArray"

File: TestVs2015/TestCpp/TestCpp.cpp
// TestCpp.cpp : Defines the entry point for the console application.
//

#include "stdafx.h"
#include <vector>
#include <list>
#include <unordered_map>
#include <map>
#include <string>

enum Gender {
	kMale,
	kFemale
};

struct Person {
	int age;
	Gender gender;
};

int wmain(int argc, wchar_t* argv[]) {
  int n = 123;
	Person p = { 2, Gender::kFemale };

  std::vector<int> v = { 10, 11, 12, 13, 14 };
  std::list<int> l = { 10, 11, 12, 13, 14 };
  std::unordered_map<int, int> u = { { 1, 11 }, { 2, 22 }, { 3, 33 } };
  std::map<int, int> m = { { 1, 11 }, { 2, 22 }, { 3, 33 } };

  std::string s1 = "hello";
  std::wstring s2 = L"world";

  char * s3 = "a small";
  wchar_t * s4 = L"good thing";

  char s5[] = "python";
  wchar_t s6[] = L"c sharp";

  return 0;
}

